pandas-stubs/io/parsers/readers.pyi

```diff
@@ -0,0 +1,438 @@
+from collections import abc
+import csv
+from types import TracebackType
+from typing import (
+    Any,
+    Callable,
+    Literal,
+    Sequence,
+    overload,
+)
+
+from pandas.core.frame import DataFrame
+from pandas.core.indexes.base import Index
+from pandas.core.series import Series
+
+import pandas._libs.lib as lib
+from pandas._typing import (
+    CompressionOptions,
+    CSVEngine,
+    DtypeArg,
+    FilePath,
+    ReadCsvBuffer,
+    StorageOptions,
+    npt,
+)
+
+from pandas.io.common import IOHandles
+
+@overload
+def read_csv(
+    filepath_or_buffer: FilePath | ReadCsvBuffer[bytes] | ReadCsvBuffer[str],
+    *,
+    sep: str | None | lib.NoDefault = ...,
+    delimiter: str | None | lib.NoDefault = ...,
+    header: int | Sequence[int] | Literal["infer"] | None = ...,
+    names: list[str] | None | lib.NoDefault = ...,
+    index_col: int | str | Sequence[str | int] | Literal[False] | None = ...,
+    usecols: list[str]
+    | tuple[str, ...]
+    | Sequence[int]
+    | Series
+    | Index
+    | npt.NDArray
+    | Callable[[str], bool]
+    | None = ...,
```

Comment:
may be missing `mangle_dupe_cols` here.  It's in the 1.4.4 docs

Reply:
This has been deprecated in 1.5.  Though we are so close might as well drop it now given how close 1.5 is.

Reply:
This is a tough call, because I don't think there is an alternate available if you were still using the feature.  Docs say "Not implemented, and a new argument to specify the pattern for the names of duplicated columns will be added instead", meaning there is no workaround.  I think we should leave it in as I will be doing a release with 1.4.4 where it is supported, and we can remove it once 1.5 is released.

Reply:
readded this.